README.md
# [Jinja2 Theme - AdminT (Free Version)](https://theme-jinja2-admint.appseed.us/)

> Template Theme provided by [AppSeed](https://appseed.us) - Features:

- UI Kit: **AdminT Dashboard** (Free Version) by **YooKits**
- Render Engine: Flask / [Jinja2](https://jinja.palletsprojects.com/)
- **[CC BY 4.0](https://creativecommons.org/licenses/by/4.0/)** (preserve the footer credit to YooKits)
- Free support via **Github** issues tracker
- Paid 24/7 Live Support via [Discord](https://discord.gg/fZC6hup).

> Links

- [Jinja2 Theme - AdminT (Free Version)](https://theme-jinja2-admint.appseed.us/) - LIVE Demo

<br />

## UI Kit - [AdminT (Free Version)](https://yookits.com/product/admint-bootstrap-admin-template/?ref=appseed)

*Vendor Notes* - **[AdminT](https://yookits.com/product/admint-bootstrap-admin-template/?ref=appseed)** is the swiss army knife of the dashboard templates. 
You can create applications of any kind with it. We carefully crafted every component based on a cohesive design system. Your app will look pro with Admint magic. It’s completely free & open-source. Grab it today and build beautiful apps.

> Links

- [AdminT (Free Version)](https://yookits.com/product/admint-bootstrap-admin-template/?ref=appseed) - product page
- [AdminT](https://github.com/yookits/admint/) - Html version source code (Github)

<br />

![Jinja2 Theme - AdminT (Free Version) - Template project provided by AppSeed.](https://raw.githubusercontent.com/app-generator/theme-jinja2-admint/master/media/theme-jinja2-admint-screen.png)

<br />

## Build from sources

```bash
$ # Clone the sources
$ git clone https://github.com/app-generator/theme-jinja2-admint.git
$ cd theme-jinja2-admint
$
$ # Virtualenv modules installation (Unix based systems)
$ virtualenv env
$ source env/bin/activate
$
$ # Virtualenv modules installation (Windows based systems)
$ # virtualenv env
$ # .\env\Scripts\activate
$
$ # Install requirements
$ pip3 install -r requirements.txt
$
$ # Set the FLASK_APP environment variable
$ (Unix/Mac) export FLASK_APP=run.py
$ (Windows) set FLASK_APP=run.py
$ (Powershell) $env:FLASK_APP = ".\run.py"
$
$ # Set up the DEBUG environment
$ # (Unix/Mac) export FLASK_ENV=development
$ # (Windows) set FLASK_ENV=development
$ # (Powershell) $env:FLASK_ENV = "development"
$
$ # Run the Jinja Template
$ # --host=0.0.0.0 - expose the app on all network interfaces (default 127.0.0.1)
$ # --port=5000    - specify the app port (default 5000)  
$ flask run --host=0.0.0.0 --port=5000
$
$ # Access the UI in browser: http://127.0.0.1:5000/
```

<br />

## Code-base structure

The project has a simple structure, represented as bellow:

```bash
< PROJECT ROOT >
   |
   |-- app/__init__.py
   |-- app/
   |    |-- static/
   |    |    |-- <css, JS, images>         # CSS files, Javascripts files
   |    |
   |    |-- templates/
   |    |    |
   |    |    |-- includes/                 # Page chunks, components
   |    |    |    |
   |    |    |    |-- navigation.html      # Top bar
   |    |    |    |-- sidebar.html         # Left sidebar
   |    |    |    |-- scripts.html         # JS scripts common to all pages
   |    |    |    |-- footer.html          # The common footer
   |    |    |
   |    |    |-- layouts/                  # App Layouts (the master pages)
   |    |    |    |
   |    |    |    |-- base.html            # Used by common pages like index, UI
   |    |    |    |-- base-fullscreen.html # Used by auth pages (login, register)
   |    |    |
   |    |  index.html                      # The default page
   |    |  login.html                      # Auth Login Page
   |    |  register.html                   # Auth Registration Page
   |    |  page-404.html                   # Error 404 page (page not found)
   |    |  page-500.html                   # Error 500 page (server error)
   |    |    *.html                        # All other pages provided by the UI Kit
   |
   |-- requirements.txt
   |
   |-- run.py
   |
   |-- ************************************************************************
```

<br />

## Deployment

The project comes with a basic configuration for [Docker](https://www.docker.com/), [Gunicorn](https://gunicorn.org/), and [Waitress](https://docs.pylonsproject.org/projects/waitress/en/stable/).

<br />

### [Docker](https://www.docker.com/) execution
---

The steps to start the template using Docker:

> Get the code

```bash
$ git clone https://github.com/app-generator/theme-jinja2-admint.git
$ cd theme-jinja2-admint
```

> Start the app in Docker

```bash
$ sudo docker-compose pull && sudo docker-compose build && sudo docker-compose up -d
```

Visit `http://localhost:5005` in your browser. The app should be up & running.

<br />

### [Gunicorn](https://gunicorn.org/)
---

Gunicorn 'Green Unicorn' is a Python WSGI HTTP Server for UNIX.

> Install using pip

```bash
$ pip install gunicorn
```
> Start the app using gunicorn binary

```bash
$ gunicorn --bind 0.0.0.0:8001 run:app
Serving on http://localhost:8001
```

Visit `http://localhost:8001` in your browser. The app should be up & running.

<br />

### [Waitress](https://docs.pylonsproject.org/projects/waitress/en/stable/)
---

Waitress (Gunicorn equivalent for Windows) is meant to be a production-quality pure-Python WSGI server with very acceptable performance. It has no dependencies except ones that live in the Python standard library.

> Install using pip

```bash
$ pip install waitress
```
> Start the app using [waitress-serve](https://docs.pylonsproject.org/projects/waitress/en/stable/runner.html)

```bash
$ waitress-serve --port=8001 run:app
Serving on http://localhost:8001
```

Visit `http://localhost:8001` in your browser. The app should be up & running.

<br />

## Credits & Links

- [Flask Framework](https://www.palletsprojects.com/p/flask/) - The official website

<br />

---
[Jinja2 Theme - AdminT (Free Version)](https://theme-jinja2-admint.appseed.us/) - Provided by **AppSeed** [Web App Generator](https://appseed.us/app-generator).


CHANGELOG.md
# Change Log

## [1.0.1] 2020-06-14
### Improvements

- Bump the UI to the latest AdminT (Free version) - 20200610 (Release Date)

## [1.0.0] 2020-06-08
### Initial Release

- Update Licensing information
- Add CHANGELOG.md to track all changes


app/static/assets/js/chartjs.light.js
$.exists = function(selector) {
  return $(selector).length > 0;
};

/* Line chart Variable */
var lineChartToolTips = {
  displayColors: false,
  mode: "nearest",
  intersect: false,
  position: "nearest",
  xPadding: 8,
  yPadding: 8,
  caretPadding: 8,
  backgroundColor: "#fff",
  cornerRadius: 4,
  titleFontSize: 13,
  titleFontStyle: "normal",
  bodyFontSize: 13,
  titleFontColor: "#222",
  bodyFontColor: "rgba(0, 0, 0, 0.7)",
  borderWidth: 1,
  borderColor: "rgba(0, 0, 0, 0.1)"
}
var scalesYaxes = [{
  ticks: {
    fontSize: 14,
    fontColor: "rgba(0, 0, 0, 0.4)",
    padding: 15,
    beginAtZero: true,
    autoSkip: false,
    maxTicksLimit: 4
  },
  gridLines: {
    color: "rgba(0, 0, 0, 0.1)",
    zeroLineWidth: 1,
    zeroLineColor: "transparent",
    drawBorder: false,
    tickMarkLength: 0
  }
}]
var scalesXaxes = [{
  ticks: {
    fontSize: 14,
    fontColor: "rgba(0, 0, 0, 0.4)",
    padding: 5,
    beginAtZero: true,
    autoSkip: false,
    maxTicksLimit: 4
  },
  gridLines: {
    display: false
  }
}];
/* End Line chart Variable */



// Type4
if ($.exists("#yoo-chart5-type4")) {
  var ctx5 = document.querySelector("#yoo-chart5-type4");
  var myChart5 = new Chart(ctx5, {
    type: "line",
    data: {
      labels: [
        "3 Dec",
        "28 Dec",
        "3 Jan",
        "8 Jan",
        "13 Jan",
        "20 Jan",
        "25 Jan"
      ],
      datasets: [{
          label: "Orders",
          data: ["0", "15", "25", "10", "5", "20", "0"],
          backgroundColor: "rgba(0, 122, 255, 0.1)",
          borderColor: "#007aff",
          borderWidth: 3,
          pointBackgroundColor: "#fff",
          pointDotRadius: 10
        },
        {
          label: "Orders",
          data: ["15", "30", "15", "30", "15", "25", "20"],
          backgroundColor: "rgba(52, 199, 89, 0.1)",
          borderColor: "#34c759",
          borderWidth: 3,
          pointBackgroundColor: "#fff",
          pointDotRadius: 10
        }
      ]
    },
    options: {
      responsive: true,
      maintainAspectRatio: false,
      legend: false,
      tooltips: lineChartToolTips,
      scales: {
        yAxes: [{
          position: "left",
          ticks: {
            fontSize: 14,
            fontColor: "rgba(0, 0, 0, 0.4)",
            padding: 15,
            autoSkip: false,
            maxTicksLimit: 6,
            beginAtZero: true,
            steps: 10,
            stepValue: 5,
            max: 40
          },
          gridLines: {
            color: "rgba(0, 0, 0, 0.1)",
            zeroLineWidth: 0,
            zeroLineColor: "transparent",
            drawBorder: false,
            tickMarkLength: 0
          }
        }],
        xAxes: scalesXaxes
      },
      elements: {
        point: {
          radius: 0
        }
      }
    }
  });
}

/* Start Line Chart3 Initialize */
if ($.exists("#yoo-chart3")) {
  var ctx3 = document.querySelector("#yoo-chart3").getContext("2d");
  var myChart3 = new Chart(ctx3, {
    type: "pie",
    data: {
      labels: ["Desktop", "Mobile", "Tablet", "Miscellaneous"],
      datasets: [{
        backgroundColor: ["#ffcc00", "#ff9500", "#ff3b30", "#5856d6"],
        hoverBackgroundColor: ["#ffcc00", "#ff9500", "#ff3b30", "#5856d6"],
        data: [60, 15, 10, 15],
        borderWidth: 0,
        hoverBorderColor: ["#ffcc00", "#ff9500", "#ff3b30", "#5856d6"],
        hoverBorderWidth: 8
      }]
    },
    options: {
      cutoutPercentage: 80,
      legend: false,
      tooltips: lineChartToolTips
    }
  });
}


app/static/assets/js/svg-maps/jquery.vmap.main.js
$.exists = function(selector) {
  return $(selector).length > 0;
};
$(document).on("ready", function() {
  $(window).trigger("resize");
  svgMap();
});

/*--------------------------------------------------------------
17. Vector Map
--------------------------------------------------------------*/
function svgMap() {

  // World Map
  if ($.exists("#world-en")) {
    $("#world-en").vectorMap({
      map: 'world_en',
      showTooltip: true,
      backgroundColor: "#fff",
      color: "rgba(88, 86, 214, 0.1)",
      colors: {
        ru: "rgba(88, 86, 214, 0.3)",
        sd: "rgba(88, 86, 214, 0.4)",
        or: "rgba(88, 86, 214, 0.5)",
        ir: "rgba(88, 86, 214, 0.6)",
        ca: "#5856d6",
        br: "rgba(88, 86, 214, 0.7)",
        au: "rgba(88, 86, 214, 0.8)"
      },
      hoverColor: "#666",
      enableZoom: true,
      borderWidth: 3,
      borderColor: "#fff",
      hoverOpacity: 0.9
    });
  }

  // USA Map
  if ($.exists("#svg-usa")) {
    $("#svg-usa").vectorMap({
      map: "usa_en",
      showTooltip: true,
      backgroundColor: "#fff",
      color: "rgba(88, 86, 214, 0.1)",
      colors: {
        mo: "rgba(88, 86, 214, 0.3)",
        fl: "rgba(88, 86, 214, 0.4)",
        or: "rgba(88, 86, 214, 0.5)",
        ca: "rgba(88, 86, 214, 0.6)",
        tx: "#5856d6",
        wy: "rgba(88, 86, 214, 0.7)",
        ny: "rgba(88, 86, 214, 0.8)"
      },
      hoverColor: "#666",
      enableZoom: true,
      borderWidth: 3,
      borderColor: "#fff",
      hoverOpacity: 0.9
    });
  }

  // Algeria Map
  if ($.exists("#dz-fr")) {
    $("#dz-fr").vectorMap({
      map: "dz_fr",
      showTooltip: true,
      backgroundColor: "#fff",
      color: "rgba(88, 86, 214, 0.1)",
      colors: {
        11: "rgba(88, 86, 214, 0.3)",
        30: "rgba(88, 86, 214, 0.4)",
        17: "rgba(88, 86, 214, 0.5)",
        32: "rgba(88, 86, 214, 0.6)",
        8: "#5856d6"
      },
      hoverColor: "#666",
      enableZoom: true,
      borderWidth: 3,
      borderColor: "#fff",
      hoverOpacity: 0.9
    });
  }

  // Argentina Map
  if ($.exists("#argentina-en")) {
    $("#argentina-en").vectorMap({
      map: "argentina_en",
      showTooltip: true,
      backgroundColor: "#fff",
      color: "rgba(88, 86, 214, 0.1)",
      hoverColor: "#666",
      enableZoom: true,
      borderWidth: 3,
      borderColor: "#fff",
      hoverOpacity: 0.9
    });
  }

  // Brazil Map
  if ($.exists("#brazil-br")) {
    $("#brazil-br").vectorMap({
      map: "brazil_br",
      showTooltip: true,
      backgroundColor: "#fff",
      color: "rgba(88, 86, 214, 0.1)",
      hoverColor: "#666",
      enableZoom: true,
      borderWidth: 3,
      borderColor: "#fff",
      hoverOpacity: 0.9
    });
  }

  // Croatia Map
  if ($.exists("#croatia-map")) {
    $("#croatia-map").vectorMap({
      map: "croatia",
      showTooltip: true,
      backgroundColor: "#fff",
      color: "rgba(88, 86, 214, 0.1)",
      hoverColor: "#666",
      enableZoom: true,
      borderWidth: 3,
      borderColor: "#fff",
      hoverOpacity: 0.9
    });
  }

  // Europe Map
  if ($.exists("#europe-en")) {
    $("#europe-en").vectorMap({
      map: "europe_en",
      showTooltip: true,
      backgroundColor: "#fff",
      color: "rgba(88, 86, 214, 0.1)",
      hoverColor: "#666",
      enableZoom: true,
      borderWidth: 3,
      borderColor: "#fff",
      hoverOpacity: 0.9
    });
  }

  // France Map
  if ($.exists("#france-fr")) {
    $("#france-fr").vectorMap({
      map: "france_fr",
      showTooltip: true,
      backgroundColor: "#fff",
      color: "rgba(88, 86, 214, 0.1)",
      hoverColor: "#666",
      enableZoom: true,
      borderWidth: 3,
      borderColor: "#fff",
      hoverOpacity: 0.9
    });
  }

  // Germany Map
  if ($.exists("#germany-en")) {
    $("#germany-en").vectorMap({
      map: "germany_en",
      showTooltip: true,
      backgroundColor: "#fff",
      color: "rgba(88, 86, 214, 0.1)",
      hoverColor: "#666",
      enableZoom: true,
      borderWidth: 3,
      borderColor: "#fff",
      hoverOpacity: 0.9
    });
  }

  // Greece Map
  if ($.exists("#greece")) {
    $("#greece").vectorMap({
      map: "greece",
      showTooltip: true,
      backgroundColor: "#fff",
      color: "rgba(88, 86, 214, 0.1)",
      hoverColor: "#666",
      enableZoom: true,
      borderWidth: 3,
      borderColor: "#fff",
      hoverOpacity: 0.9
    });
  }

  // Greece Map
  if ($.exists("#indonesia_id")) {
    $("#indonesia_id").vectorMap({
      map: "indonesia_id",
      showTooltip: true,
      backgroundColor: "#fff",
      color: "rgba(88, 86, 214, 0.1)",
      hoverColor: "#666",
      enableZoom: true,
      borderWidth: 3,
      borderColor: "#fff",
      hoverOpacity: 0.9
    });
  }

  // Iran Map
  if ($.exists("#iran_ir")) {
    $("#iran_ir").vectorMap({
      map: "iran_ir",
      showTooltip: true,
      backgroundColor: "#fff",
      color: "rgba(88, 86, 214, 0.1)",
      hoverColor: "#666",
      enableZoom: true,
      borderWidth: 3,
      borderColor: "#fff",
      hoverOpacity: 0.9
    });
  }

  // Iran Map
  if ($.exists("#iraq")) {
    $("#iraq").vectorMap({
      map: "iraq",
      showTooltip: true,
      backgroundColor: "#fff",
      color: "rgba(88, 86, 214, 0.1)",
      hoverColor: "#666",
      enableZoom: true,
      borderWidth: 3,
      borderColor: "#fff",
      hoverOpacity: 0.9
    });
  }

  // Turkey Map
  if ($.exists("#turkey")) {
    $("#turkey").vectorMap({
      map: "turkey",
      showTooltip: true,
      backgroundColor: "#fff",
      color: "rgba(88, 86, 214, 0.1)",
      hoverColor: "#666",
      enableZoom: true,
      borderWidth: 3,
      borderColor: "#fff",
      hoverOpacity: 0.9
    });
  }

  // Ukraine Map
  if ($.exists("#ukraine_ua")) {
    $("#ukraine_ua").vectorMap({
      map: "ukraine_ua",
      showTooltip: true,
      backgroundColor: "#fff",
      color: "rgba(88, 86, 214, 0.1)",
      hoverColor: "#666",
      enableZoom: true,
      borderWidth: 3,
      borderColor: "#fff",
      hoverOpacity: 0.9
    });
  }

}


app/static/assets/js/main.js
(function($) {
  "use strict";

  /*--------------------------------------------------------------
    Scripts initialization
  --------------------------------------------------------------*/

  $(window).on("load", function() {
    $(window).trigger("scroll");
    $(window).trigger("resize");
  });

  $(document).on("ready", function() {
    $(window).trigger("resize");
    mainMenu();
    mobileMenu();
    sideBarHeader();
    stickyHeader();
    onePageNav();
    dynamicBackgroundImage();
    tabInt();
    accordianInt();
    progressBar();
    calendarInt();
    customSelectInt();
    modalVideoInt();
    nicescrollInt();
    customToggleInt();
    emailEditorInt();
    nestableSetup();
    dynamicTableInt();
    bootstrapInt();
  });

  $(window).on("resize", function() {
    mobileMenu();
  });

  $(window).on("scroll", function() {
    stickyHeader();
  });

  $.exists = function(selector) {
    return $(selector).length > 0;
  };

  /*--------------------------------------------------------------
    1. Header Sctipt
  --------------------------------------------------------------*/

  /* Main Menu */
  function mainMenu() {
    $(".yoo-nav-toggle").on("click", function() {
      $(this).siblings(".yoo-nav").slideToggle();
      $(this).toggleClass("yoo-active");
    });
    $(".yoo-has-children").append('<span class="yoo-dropdown-btn"></span>');
    $(".yoo-dropdown-btn").on("click", function() {
      $(this).siblings("ul, .yoo-megamenu-in, .yoo-vertical-megamenu-in").slideToggle("slow");
      $(this).toggleClass("yoo-active");
    });

  }
  /* Mobile Menu */
  function mobileMenu() {
    if ($(window).width() <= 991) {
      $(".yoo-header").addClass("yoo-mobile-header");
      $(".yoo-nav").addClass("yoo-mobile-nav").removeClass("yoo-desktop-nav");
    } else {
      $(".yoo-header").removeClass("yoo-mobile-header");
      $(".yoo-nav").addClass("yoo-desktop-nav").removeClass("yoo-mobile-nav");
    }
  }
  /* Sticky Header */
  function stickyHeader() {
    var scroll = $(window).scrollTop();
    if (scroll >= 10) {
      $(".yoo-header").addClass("yoo-sticky-active");
    } else {
      $(".yoo-header").removeClass("yoo-sticky-active");
    }
  }
  /* Sidebar Header */
  function sideBarHeader() {
    $(".yoo-sidebar-has-children").append('<span class="yoo-dropdown-arrow"></span>');
    $(".yoo-sidebar-has-children>a").removeAttr("href").on("click", function() {
      $(this).siblings(".yoo-sidebar-nav-dropdown").slideToggle();
      $(this).siblings(".yoo-dropdown-arrow").toggleClass("active");
    });
    $(".yoo-sidebarheader-toggle").on("click", function() {
      $("body").toggleClass("yoo-sidebar-active");
    });
    // Hover To Class Toggle
    $(".yoo-sidebarheader").hover(
      function() {
        $("body").addClass("yoo-sidebar-hover-active");
      },
      function() {
        $("body").removeClass("yoo-sidebar-hover-active");
      }
    );
  }
  /*--------------------------------------------------------------
    2. One Page Navigation
  --------------------------------------------------------------*/
  function onePageNav() {
    $('.yoo-doc-nav-dropdown-btn').on('click', function() {
      $(this).toggleClass('active');
      $(this).siblings('.yoo-doc-nav-dropdown-list').slideToggle();
    })
    // Click to Go Top Animation
    $('.yoo-onepage-nav a').each(function() {
      var thisAttr = $(this).attr('href');
      if ($(thisAttr).length) {
        $(this).addClass('yoo-page-nav');
      }
    });
    $('.yoo-page-nav').on('click', function() {
      var thisAttr = $(this).attr('href');
      if ($(thisAttr).length) {
        var scrollPoint = $(thisAttr).offset().top - 80;
        $('body,html').animate({
          scrollTop: scrollPoint
        }, 650);
      }
      return false;
    });
    // One Page Active Class
    var topLimit = 300,
      ultimateOffset = 200;
    $('.yoo-onepage-nav').each(function() {
      var $this = $(this),
        $parent = $this.parent(),
        current = null,
        $findLinks = $this.find("a");

      function getHeader(top) {
        var last = $findLinks.first();
        if (top < topLimit) {
          return last;
        }
        for (var i = 0; i < $findLinks.length; i++) {
          var $link = $findLinks.eq(i),
            href = $link.attr("href");
          if (href.charAt(0) === "#" && href.length > 1) {
            var $anchor = $(href).first();
            if ($anchor.length > 0) {
              var offset = $anchor.offset();
              if (top < offset.top - ultimateOffset) {
                return last;
              }
              last = $link;
            }
          }
        }
        return last;
      }
      $(window).on("scroll", function() {
        var top = window.scrollY,
          height = $this.outerHeight(),
          max_bottom = $parent.offset().top + $parent.outerHeight(),
          bottom = top + height + ultimateOffset;
        var $current = getHeader(top);
        if (current !== $current) {
          $this.find(".active").removeClass("active");
          $current.addClass("active");
          current = $current;
        }
      });
    });
  }

  /*--------------------------------------------------------------
    3. Background Image Script
  --------------------------------------------------------------*/
  function dynamicBackgroundImage() {
    $('.yoo-dynamicbg').each(function() {
      var src = $(this).attr('data-src');
      $(this).css({
        'background-image': 'url(' + src + ')'
      });
    });
  }

  /*--------------------------------------------------------------
    4. Tab Style
  --------------------------------------------------------------*/
  function tabInt() {
    $('.yoo-tabs.yoo-fade-tabs .yoo-tab-links a:not(.yoo-work-link)').on('click', function(e) {
      var currentAttrValue = $(this).attr('href');
      $('.yoo-tabs ' + currentAttrValue).fadeIn(400).siblings().hide();
      $(this).parents('li').addClass('yoo-active').siblings().removeClass('yoo-active');
      e.preventDefault();
    });
  }

  /*--------------------------------------------------------------
    4. Accordion
  --------------------------------------------------------------*/
  function accordianInt() {
    $('.yoo-accordian.yoo-active').find('.yoo-accordian-title').addClass('yoo-open');
    $(".yoo-accordian-wrap .yoo-accordian-title").on('click', function(e) {
      $(this).toggleClass('yoo-open');
      $(this).parents('.yoo-accordian').siblings().find('.yoo-accordian-title').removeClass('yoo-open');
      $(this).parents('.yoo-accordian-wrap').find(".yoo-accordian-body").slideUp();
      if ($(this).next().is(":hidden")) {
        $(this).next().slideDown();
      }
    });
  }

  /*--------------------------------------------------------------
    5. Progress Bar
  --------------------------------------------------------------*/
  function progressBar() {
    $(".yoo-single-progress").each(function() {
      let progressPercentage = $(this).find(".yoo-progressbar").data("progress-percentage") + "%";
      $(this).find(".yoo-progressbar-in").css("width", progressPercentage);
      $(this).find(".yoo-progress-percentage").html(progressPercentage);
    });
  }

  /*--------------------------------------------------------------
    6. Calendar
  --------------------------------------------------------------*/
  function calendarInt() {
    if ($.exists("#yoo-calendar")) {
      $('#yoo-calendar').fullCalendar({
        header: {
          left: 'prev,next today',
          center: 'title',
          right: 'month,basicWeek,basicDay'
        },
        defaultDate: '2019-01-12',
        navLinks: true, // can click day/week names to navigate views
        editable: true,
        eventLimit: true, // allow "more" link when too many events
        events: [{
            title: 'All Day Event',
            start: '2019-01-01T20:00:00'
          },
          {
            title: 'Long Event',
            start: '2019-01-07T10:00:00',
            end: '2019-01-08T16:00:00'
          },
          {
            title: 'Repeating Event',
            start: '2019-01-10T16:00:00'
          },
          {
            title: 'Conference',
            start: '2019-01-10T09:00:00'
          },
          {
            title: 'Meeting',
            start: '2019-01-12T10:30:00',
            end: '2019-01-12T12:30:00'
          },
          {
            title: 'Lunch',
            start: '2019-01-12T12:00:00'
          },
          {
            title: 'Meeting',
            start: '2019-01-12T14:30:00'
          },
          {
            title: 'Happy Hour',
            start: '2019-01-12T17:30:00'
          },
          {
            title: 'Dinner',
            start: '2019-01-01T14:00:00',
            end: '2019-01-02T14:30:00'
          },
          {
            title: 'Birthday Party',
            start: '2019-01-13T07:00:00'
          },
          {
            title: 'Click for Google',
            url: 'http://google.com/',
            start: '2019-01-28T08:00:00'
          }
        ]
      });
    }
  }

  /*--------------------------------------------------------------
    7. Custom Select
  --------------------------------------------------------------*/
  function customSelectInt() {
    $(".yoo-custom-select").each(function() {
      var classes = $(this).attr("class"),
        id = $(this).attr("id"),
        name = $(this).attr("name");
      var template = '<div class="' + classes + '">';
      template += '<span class="custom-select-trigger">' + $(".yoo-custom-select-wrap > .yoo-custom-select option:first").html() + "</span>";
      template += '<div class="custom-options">';
      $(this).find("option").each(function() {
        template += '<span class="custom-option ' + $(this).attr("class") + " data-value=" + $(this).attr("value") + '">' + $(this).html() + "</span>";
      });
      template += "</div></div>";

      $(this).wrap('<div class="custom-select-wrapper"></div>');
      $(this).hide();
      $(this).after(template);
    });
    $(".custom-select-trigger").on("click", function(event) {
      $("html").one("click", function() {
        $(".yoo-custom-select").removeClass("opened");
      });
      $(this).parents(".yoo-custom-select").toggleClass("opened");
      event.stopPropagation();
    });
    $(".custom-option").on("click", function() {
      $(this).parents(".custom-select-wrapper").find("select").val($(this).data("value"));
      $(this).parents(".custom-options").find(".yoo-custom-option").removeClass("selection");
      $(this).addClass("selection");
      $(this).parents(".custom-select").removeClass("opened");
      $(this).parents(".yoo-custom-select").find(".custom-select-trigger").text($(this).text());
    });
  }


  /*--------------------------------------------------------------
    10. Modal Video
  --------------------------------------------------------------*/
  function modalVideoInt() {
    $(document).on('click', '.yoo-video-open', function(e) {
      e.preventDefault();
      var video = $(this).attr('href');
      $('.yoo-video-popup-container iframe').attr('src', video);
      $('.yoo-video-popup').addClass('active');

    });
    $('.yoo-video-popup-close, .yoo-video-popup-layer').on('click', function(e) {
      $('.yoo-video-popup').removeClass('active');
      $('html').removeClass('overflow-hidden');
      $('.yoo-video-popup-container iframe').attr('src', 'about:blank')
      e.preventDefault();
    });
  }

  /*--------------------------------------------------------------
    12. Nice Scroll
  --------------------------------------------------------------*/
  function nicescrollInt() {
    Scrollbar.initAll();
  }

  /*--------------------------------------------------------------
    14. Toggle Btn
  --------------------------------------------------------------*/
  function customToggleInt() {
    // Custome Toggle Button
    $(".yoo-toggle-btn").on("click", function() {
      $(this).toggleClass("active").siblings(".yoo-dropdown").toggleClass("active");
      $(this).parents("li").siblings().find(".yoo-dropdown, .yoo-toggle-btn").removeClass("active");
      $(this).parents('.yoo-toggle-body').siblings().find('.yoo-dropdown, .yoo-toggle-btn').removeClass('active');
    });
    $('.yoo-toggle-cross-btn').on('click', function() {
      $(this).parents('.yoo-toggle-body').find('.yoo-toggle-btn, .yoo-dropdown').removeClass('active');
    })
    $(document).on("click", function() {
      $(".yoo-dropdown").removeClass("active").siblings().removeClass("active");
    });
    $(".yoo-dropdown, .yoo-toggle-btn").on("click", function(e) {
      e.stopPropagation();
    });

    $(".yoo-circle-color, .yoo-vertical-chart-layer").each(function() {
      let buletColor = $(this).data("bulet-color");
      $(this).css("background-color", buletColor);
    });
    $('.yoo-toggle-cross').on('click', function(){
      $(this).parents('.yoo-dropdown').removeClass('active').siblings('.yoo-toggle-btn').removeClass('active');
    })
    // Star Toggle Btn
    $('.yoo-get-star, .yoo-mobile-toggle-btn').on('click', function() {
      $(this).toggleClass('active');
    })

    // Switch Btn Toggle
    $('.yoo-switch').on('click', function() {
      $(this).toggleClass('yoo-active');
    })
    // Profile Toggle
    $('.yoo-profile-sidebar-btn').on('click', function() {
      $(this).parents('.yoo-profile-sidebar').addClass('active');
    })
    $('.yoo-profile-sidebar-cross').on('click', function() {
      $(this).parents('.yoo-profile-sidebar').removeClass('active');
    })
  }

  /*--------------------------------------------------------------
    15. Email Editor
  --------------------------------------------------------------*/
  function emailEditorInt() {
    if ($.exists("#yoo-email-editor")) {
      console.clear()
      var FontAttributor = Quill.import('formats/font');
      var fonts = ['impact', 'courier', 'comic'];
      var lHeights = ['1.0', '1.1', '1.2', '1.3', '1.4', '1.5', '1.6'];
      FontAttributor.whitelist = fonts;
      Quill.register(FontAttributor, true);
      var quill = new Quill('#yoo-email-editor', {
        modules: {
          toolbar: { container: '#yoo-email-editor-toolbox' }
        },
        placeholder: 'Type something here',
        theme: 'snow' // 'snow' or 'bubble'
      });
      $('.ql-bold').html('<i class="material-icons-outlined">format_bold</i>');
      $('.ql-italic').html('<i class="material-icons-outlined">format_italic</i>');
      $('.ql-underline').html('<i class="material-icons-outlined">format_underlined</i>');
    }
  }

  /*--------------------------------------------------------------
    16. Nestable List
  --------------------------------------------------------------*/
  function nestableSetup() {
    if ($.exists("#yoo-nestable1")) {
      $('#yoo-nestable1').nestable();
    }
    if ($.exists("#yoo-nestable2")) {
      $('#yoo-nestable2').nestable();
    }

    $('.yoo-nested-dropdown-btn').on('click', function() {
      $(this).toggleClass('yoo-active').siblings('.yoo-nested-dropdown').slideToggle();
    });
    $('.yoo-nosted-cancel').on('click', function() {
      $(this).parents('.yoo-nested-dropdown').slideUp();
    })
    $('.yoo-nosted-delete').on('click', function() {
      $(this).parents('.yoo-nested-dropdown-wrap').parent('.dd-item').remove();
    })
  }

  /*--------------------------------------------------------------
    17. Dynamic Table
  --------------------------------------------------------------*/
  function dynamicTableInt() {
    $("#yoo-left-locked").each(function() {
      var table = $('#yoo-left-locked').DataTable({
        scrollY: "600px",
        scrollX: true,
        scrollCollapse: true,
        paging: false,
        fixedColumns: {
          leftColumns: 1
        }
      });
    })
    $("#yoo-right-locked").each(function() {
      var table = $('#yoo-right-locked').DataTable({
        scrollY: "600px",
        scrollX: true,
        scrollCollapse: true,
        paging: false,
        fixedColumns: {
          leftColumns: 0,
          rightColumns: 1
        }
      });
    })
    $("#yoo-left-right-locked").each(function() {
      var table = $('#yoo-left-right-locked').DataTable({
        scrollY: "600px",
        scrollX: true,
        scrollCollapse: true,
        paging: false,
        fixedColumns: {
          leftColumns: 1,
          rightColumns: 1
        }
      });
    })
    $("#yoo-no-locked").each(function() {
      var table = $('#yoo-no-locked').DataTable({
        scrollX: true,
        scrollCollapse: true,
        paging: false
      });
    })
  }

  /*--------------------------------------------------------------
    18. Bootstrap Setup Mode
  --------------------------------------------------------------*/
  function bootstrapInt() {
    $('.your-checkbox').prop('indeterminate', true);

    $('#myModal').on('shown.bs.modal', function() {
      $('#myInput').trigger('focus')
    })
  }


})(jQuery); // End of use strict
